src/Microsoft.Diagnostics.Runtime/ClrValue.cs

```diff
@@ -0,0 +1,1077 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Runtime
+{
+    /// <summary>
+    /// A ClrValue represents the value of a field or variable.  This value may be of any type (or null).
+    /// </summary>
+    public abstract class ClrValue
```

Comment:
In general this came out of a realization that you always do the following every time you enumerate the heap:

    foreach (ulong address in heap.EnumerateObjectAddresses())
    {
        ClrType type = heap.GetObjectType(address);  //  <-  This is basically required code.

Having them wrapped up into one struct of two pointers in size is just more efficient both size-wise (structs are generally faster when they are 16 bytes or smaller) and in the sense that the user *has* to get the type to do anything meaningful with the object at all.

As for inheritance/interface, I agree I probably need to add an interface here, but I didn't want to add one until I have some perf data on how much it will cost.  (The answer is probably nothing, performance related...or at least so small it doesn't matter.)  I plan to look into that this week if I have time.